uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -411,9 +412,15 @@ private static function validateLeaveDatesDoesNotOverlapContracts($params) {
       'end_date' => $toDate->format('Y-m-d'),
     ]);
 
-    if ($contract['count'] > 1) {
+    $contractForEndDate = civicrm_api3('HRJobContract', 'getcontractswithdetailsinperiod', [
```

Comment:
Well it will always be a singular contract because the end date can only be in one contract. Whether the toDate is in the past or not, the contract containing it will be returned so it won't pass the validation